x-pack/plugins/enterprise_search/server/lib/check_access.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import AbortController from 'abort-controller';
+import fetch from 'node-fetch';
+
+import { KibanaRequest, Logger } from 'src/core/server';
+import { SecurityPluginSetup } from '../../../security/server';
+import { ServerConfigType } from '../plugin';
+
+interface ICheckAccess {
+  enterpriseSearchPath: 'as' | 'ws';
+  request: KibanaRequest;
+  security?: SecurityPluginSetup;
+  config: ServerConfigType;
+  log: Logger;
+}
+type TTimeout = ReturnType<typeof setTimeout>;
+
+/**
+ * Determines whether the user has access to our Enterprise Search products
+ * via HTTP call. If not, we hide the corresponding plugin links from the
+ * nav and catalogue in `plugin.ts`, which disables plugin access
+ */
+export const checkAccess = async ({
+  enterpriseSearchPath,
+  config,
+  security,
+  request,
+  log,
+}: ICheckAccess): Promise<boolean> => {
+  // If security has been disabled, always show the plugin
+  if (!security?.authz?.mode.useRbacForRequest(request)) {
+    return true;
+  }
+
+  // If the user is a superuser or the enterprise_search user, always show the plugin
+  const isSuperUser = async () => {
+    try {
+      const user = security.authc.getCurrentUser(request);
+      return user?.roles.includes('superuser') || user?.username === 'enterprise_search';
```

Comment:
I don't really know if this is best way to determine a "superuser" but the `checkPrivilegesWithRequest` method being used in the original POC simply wasn't working for me in elasticsearch-native auth. All users that I created maintained "admin" access no matter what 😞

Reply:
Resolved via email thread - Brandon and Larry showing me how to correctly set up a role/user 😛